### PR TITLE
Feature/app framework and executable file

### PR DESCRIPTION
**Pull request by rocky02**
### Summary : 
This branch deals with building a framework for the application and an executable to start the application along with a set of arguments which the application requires as mentioned in the README.md file.

##### Changing the execution command of q_it :
Initially 
```ruby app.rb <service-name> <url/name> <sleep_period>```

Now 
```bin/q_it_server <service-name> <url/name> <sleep_period>```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
-aws.yml
+# Ignore Files
 .byebug_history
 
+# Ignore Folders
+config/aws.*
```
Comment (glen) [inline .gitignore]: newline

File: bin/app.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 require 'rubygems'
 require 'bundler/setup'
-require_relative 'publisher.rb'
-require_relative 'subscriber.rb'
-require_relative 'q_it_errors'
+require_relative '../lib/publisher.rb'
+require_relative '../lib/subscriber.rb'
+require_relative '../lib/q_it_errors'
```
Comment (glen) [inline bin/app.rb]: You could either put these in an array and loop through the array - requiring each one. 
OR it you want all the files then maybe `Dir.glob(File.join('lib', '**', '*.rb'))` ?

File: lib/load_aws.rb [hunk 3]
```diff
@@ -2,14 +2,14 @@
 
 def generate_aws_yml_file
   sample_aws = {"aws"=>{"access_key_id"=>"", "secret_access_key"=>"", "region"=>""}}
-  File.open(File.join(Dir.pwd, 'aws.yml'), 'w') do |file|
+  File.open(File.join(Dir.pwd, 'config/aws.yml'), 'w') do |file|
```
Comment (glen) [inline lib/load_aws.rb]: How about having the path to the aws.yml file in a constant and then use it everywhere?

Comment (glen) [inline lib/load_aws.rb]: I'd prefer if you have something akin to `Rails.root` - that will get our bearings right when we are trying to navigate or load files from folders.

**Review by glen — CHANGES_REQUESTED** [review]
Looking good. Few changes here and there to make things simpler in the long run.

File: lib/load_aws.rb (hunk 3 above)
Comment (rocky02) [reply lib/load_aws.rb]: Sounds good. Thinking of making a class QIt, where we can add a class method root which will return the application root path.. also going forward, anything else that can be used throughout the application can be put in here. That's the general idea. Thoughts?

File: config/q_it.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# This class contains methods and constants to be used throughout 
+# the application among all classes and modules.
+class QIt
+ 
+  def self.root
+    Dir.pwd
+  end
+end
```
Comment (glen) [inline config/q_it.rb]: Newline

File: config/q_it.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# This class contains methods and constants to be used throughout 
+# the application among all classes and modules.
+class QIt
+ 
+  def self.root
```
Comment (glen) [inline config/q_it.rb:7]: Good.
Just make changes everywhere - use `QIt.root` as the base to all other locations.

Comment (rocky02) [reply config/q_it.rb:7]: I think I have made those changes already. Do you see anywhere else we could use it?

File: config/q_it.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+    Dir.pwd
```
Comment (glen) [inline config/q_it.rb]: Instead of using `Dir.pwd` each time - how about `@@root ||= Dir.pwd` ?

File: lib/load_aws.rb [hunk 7]
```diff
@@ -1,15 +1,18 @@
...
-  File.open(File.join(Dir.pwd, 'aws.yml'), 'w') do |file|
+  File.open(AWS_PATH, 'w') do |file|
     file.puts sample_aws.to_yaml
   end
 end
 
 begin
-  if File.exists?(File.join(Dir.pwd, 'aws.yml'))
-    AWS = YAML.load(File.read('aws.yml'))["aws"]
+  if File.exists?(File.join(Dir.pwd, 'config/aws.yml'))
```
Comment (glen) [inline lib/load_aws.rb]: Use `AWS_PATH` here.

Comment (rocky02) [reply lib/load_aws.rb]: Have already done that.. let me check in case I forgot to add and push that file..

**Review by glen — APPROVED** [review]

**Pull request merged**